### DISCUSSION TITLE
Does SwiftagramCrypto uses Exempt Encryption?

### DISCUSSION THREAD

**Discussion by emreond** [sbertix/Swiftagram#165] [Q&A] [answered]
Hello,

When I sending my app to testflight/appstore as you know It asks me whether my app uses encryption or not. I answered that question to NO. I also set  ITSAppUsesNonExemptEncryption property in my app to false because I haven't implement any custom encryption logic. Should I set it to true because of SwiftagramCrypto framework?

**P.S:** As I looked the source codes I saw custom encryption methods (as expected) but I am not sure when to set that plist value to YES.

Swiftagram Version: 4.2.10

**Comment by sbertix** [comment, marked as answer]
Hey @emreond, 

No it does not uses exempt encryption: it's in another target precisely because it should be disclosed. 
I know you're using `Endpoint.Friendship.follow(_:)`, as it's in your other discussion, and unfortunately that uses non-exempt encryption in order to work. 
Please keep in mind if you only want to rely on `KeychainStorage`, though, that is considered exempt so all you have to do is importing **Swiftagram** and **ComposableRequestCrypto**, separately, instead of **SwiftagramCrypto**.

**Reply by emreond** [reply]
As you mentioned, I am using follow endpoint so in order work I need to implement `SwiftagramCrypto`. Should I set `ITSAppUsesNonExemptEncryption ` to YES when I import this framework?

**Reply by sbertix** [reply]
Precisely.
